dmgmodify/dmgmodify/modify.go

```diff
@@ -0,0 +1,163 @@
+package dmgmodify
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"strings"
+
+	"github.com/mozilla-services/stubattribution/dmglib"
+	"github.com/vimeo/go-util/crc32combine"
+)
+
+var (
+	ErrSentinelMissing        = errors.New("dmgmodify: sentinel value not found")
+	ErrBlkxResNotFound        = errors.New("dmgmodify: unable to find blkx resource to update")
+	ErrCodeTooLong            = errors.New("dmgmodify: attribution code is too long")
+	TAB                       = 0x9
+	NUL                       = 0x0
+	crcPolynomial      uint32 = 0xedb88320
+)
+
+// Update `dmg`, replacing the `sentinel` area with the provided `code`.
+// This function is a port of the C implementation from libdmg-hfsplus
+// (https://github.com/mozilla/libdmg-hfsplus/blob/a0a959bd25370c1c0a00c9ec525e3e78285adbf9/dmg/attribution.c#L209)
+// Note: We explicitly do _not_ update the Attribution resource here, as it
+// not necessary, and makes for unnecessary work in the critical path of a
+// new Firefox install. This does not impact the attribution of the build, but
+// it does mean the build cannot be re-attributed later (which is not something
+// that ever needs to happen).
+func WriteAttributionCode(dmg *dmglib.DMG, sentinel string, code []byte) error {
+	// First, pull the information we need to update the attribution code.
+	// The blkx resource has some metadata that we need to update after
+	// injecting the attribution code.
+	blkxRes, _ := dmg.Resources.GetResourceDataByName("blkx")
```

Comment:
I am wondering why we silently swallow the error here? Compared to, e.g.:

```suggestion
	blkxRes, err := dmg.Resources.GetResourceDataByName("blkx")
	if err != nil {
	  return err
	}
```

Reply:
No good reason I can think of...I probably forgot to update this after some early testing/hacking.